SofaKernel/modules/SofaBaseVisual/VisualModelImpl.cpp

```diff
@@ -285,6 +285,23 @@ bool VisualModelImpl::hasOpaque()
 
 void VisualModelImpl::drawVisual(const core::visual::VisualParams* vparams)
 {
+    if (d_componentState.getValue() == sofa::core::objectmodel::ComponentState::Loading)
+    {
+        if (m_topoChanged)
+        {
+            deleteBuffers();
+            m_topoChanged = false;
+        }
+        if (m_textureChanged)
+        {
+            deleteTextures();
+            m_textureChanged = false;
+        }
+        init();
+        initVisual();
+        updateBuffers();
+        d_componentState = sofa::core::objectmodel::ComponentState::Valid;
```

Comment:
So if any texture or anything topological is modified, everything is wiped out and built again, right ?🤔
The texture is not updating that much I guess so I would say that's not really a problem (even if the buffers and textures are totally not related). As for the topology, I suppose there will a performance hit when there is a lot of topological changes happening, like carving/cutting/etc.
And wont this code defeat the purpose of handleTopologyChange()?🤨 (if there is a new topology each time, I guess it is not needed 😆)

Reply:
@fredroy is right for pointing the performance issue. My point of view is that:
1 - behavior should always be valid and consistant (this is an hard constraint)
2 - then within the boundary of a consistent behavior then optimiz (a smooth constraint :)). 

Don't do the reverse way :)